hudi-common/src/main/java/org/apache/hudi/common/model/HoodieRecord.java

```diff
@@ -195,6 +206,10 @@ public HoodieKey getKey() {
     return key;
   }
 
+  public boolean isPartial() {
+    return isPartial;
```

Comment:
Not sure why we store `isPartial` flag in row-level, from high level, the `isPartial` should at lest to be with glanularity of commit/a write batch level.

Reply:
`isPartial` is determined at the commit or write batch level, but for record merging to work in the current implementation and maintain the layering, it's better to have the flag at the record level.

Reply:
-1, does not make sense

Reply:
I removed all changes to the `HoodieRecord` and subclasses.  Now whether a record is partial or not is determined by the schema attached, which is per log file.  Checking whether a schema is partial or not also leverages cache (see `SparkPartialMergingUtils`), so there is no overhead.